pointerevents/pointerevent_support.js

```diff
@@ -185,3 +193,49 @@ function rPointerCapture(e) {
     catch(e) {
     }
 }
+
+var globalPointerEventTest = null;
+var expectedPointerType = null;
+
+var HOVERABLE_POINTERS = ['mouse', 'pen'];
+
+function MultiPointerTypeTest(testName, types) {
+    this.testName = testName;
+    this.types = types;
+    this.currentTypeIndex = 0;
+    this.createNextTest();
+}
+
+MultiPointerTypeTest.prototype.skip = function() {
+    this.createNextTest().timeout();
+}
+
+MultiPointerTypeTest.prototype.done = function() {
+    this.createNextTest().done();
+}
+
+MultiPointerTypeTest.prototype.createNextTest = function() {
+    var prevTest = this.currentTest;
+    if (this.currentTypeIndex < this.types.length) {
+       var pointerTypeDescription = document.getElementById('pointerTypeDescription');
+       document.getElementById('pointerTypeDescription').innerHTML = "Follow the test instructions with <span style='color: red'>" + this.types[this.currentTypeIndex] + "</span>. If you don't have the device <a href='javascript:;' onclick='globalPointerEventTest.skip()'>skip it</a>.";
+       this.currentTest = async_test(this.types[this.currentTypeIndex] + ' ' + this.testName);
+       expectedPointerType = this.types[this.currentTypeIndex];
+       this.currentTypeIndex++;
+    } else {
+        document.getElementById('pointerTypeDescription').innerHTML = "";
+    }
+   resetTestState();
+   return prevTest;
+}
+
+
+function setup_pointerevent_test(testName) {
+    if (typeof supported_pointerTypes != 'undefined') {
```

Comment:
Rather than rely on global variables for communication between different source files, I think it would be easier to understand if you passed the data directly.  Eg. how about having setup_pointerevent_test take an optional 2nd arg that's the list of supported pointer types?